src/Data/CurrentUser/getWhoAmI.ts

```diff
@@ -0,0 +1,106 @@
+import { createRestApiClient, DataConnectionError } from 'scrivito'
+import { isOptionalString } from '../../utils/isOptionalString'
+import { pisaConfig, pisaUrl } from '../pisaClient'
+import { errorToast } from './errorToast'
+
+export async function getWhoAmI(): Promise<WhoAmI | null> {
+  const whoAmIConfig = await pisaConfig('whoami') // TODO: switch back to pisaClient, once #11616 is resolved
+  if (!whoAmIConfig) return null
+
+  try {
+    const whoAmI = await requestPisa(whoAmIConfig.url, whoAmIConfig.headers)
+    if (!isWhoAmI(whoAmI)) throw new DataConnectionError('Invalid user ID')
+
+    return whoAmI
+  } catch (error) {
+    errorToast('Failed to fetch whoami', error)
+    throw error
+  }
+}
+
+export async function verifySameWhoAmIUser(
+  email: string,
+  pisaAuthorization: string | null,
+) {
+  if (!pisaAuthorization) return
+
+  const baseUrl = await pisaUrl()
+  if (!baseUrl) return
+
+  const response = await fetch(`${baseUrl}/whoami`, {
```

Comment:
* [ ] FYI Should there be a TODO comment here to use the SDK helper when it supports custom auth?